jumanji/training/networks/macvrp/actor_critic.py

```diff
@@ -0,0 +1,299 @@
+                w_init_scale=1 / self.num_customers,
+                model_size=self.model_size,
+                name=f"self_attention_customer_block_{block_id}",
+            )
+            embeddings = transformer_block(
+                query=embeddings,
+                key=embeddings,
+                value=embeddings,
+            )
+
+            # Cross attention between the customer's ops embedding and vehicle embedding
+            transformer_block = TransformerBlock(
+                num_heads=self.transformer_num_heads,
+                key_size=self.transformer_key_size,
+                mlp_units=self.transformer_mlp_units,
+                w_init_scale=1 / self.num_customers,
+                model_size=self.model_size,
+                name=f"cross_attention_ops_vehicles_block_{block_id}",
+            )
+            embeddings = transformer_block(
+                query=embeddings,
+                key=v_embedding,
+                value=v_embedding,
+            )
+
+        return embeddings
+
+    def self_attention_joint_vehicles_customers(
+        self, embeddings: chex.Array
+    ) -> chex.Array:
+        for block_id in range(self.num_customers):
+            transformer_block = TransformerBlock(
+                num_heads=self.transformer_num_heads,
+                key_size=self.transformer_key_size,
+                mlp_units=self.transformer_mlp_units,
+                w_init_scale=1 / self.num_customers,
+                model_size=self.model_size,
+                name=f"self_attention_joint_ops_vehicles_block_{block_id}",
+            )
+            embeddings = transformer_block(
+                query=embeddings,
+                key=embeddings,
+                value=embeddings,
+            )
+        return embeddings
+
+
+def make_actor_network_macvrp(
+    num_vehicles: int,
+    num_customers: int,
+    transformer_num_heads: int,
+    transformer_key_size: int,
+    transformer_mlp_units: Sequence[int],
+) -> FeedForwardNetwork:
+    def network_fn(observation: Observation) -> chex.Array:
+        torso = MACVRPTorso(
+            num_vehicles=num_vehicles,
+            num_customers=num_customers,
+            transformer_num_heads=transformer_num_heads,
+            transformer_key_size=transformer_key_size,
+            transformer_mlp_units=transformer_mlp_units,
+            name="actor_torso",
+        )
+        embeddings = torso(observation)  # (B, V+C+1, D)
+
+        vehicle_embeddings, customer_embeddings = jax.numpy.split(
+            embeddings, (num_vehicles,), axis=-2
+        )
+
+        vehicle_embeddings = hk.Linear(32, name="policy_head_vehicles")(
+            vehicle_embeddings
+        )
+
+        logits = jax.numpy.einsum(
+            "...vk,...ck->...vc", vehicle_embeddings, customer_embeddings
+        )  # (B, V, C+1)
+
+        customer_embeddings = hk.Linear(32, name="policy_head_customers")(
+            customer_embeddings
+        )
```

Comment:
These two lines should be swapped, right?
```suggestion
        customer_embeddings = hk.Linear(32, name="policy_head_customers")(
            customer_embeddings
        )

        logits = jax.numpy.einsum(
            "...vk,...ck->...vc", vehicle_embeddings, customer_embeddings
        )  # (B, V, C+1)
```

Reply:
Nice catch :raised_hands: